pkg/abi/linux/fuse.go

```diff
@@ -262,6 +263,63 @@ type FUSEInitOut struct {
 	_ [8]uint32
 }
 
+// FUSEInitRes is a variable-length wrapper of FUSEInitOut. The FUSE server may
+// implement older version of FUSE protocol, which contains a FUSEInitOut with
+// less attributes.
+//
+// Dynamically-sized objects cannot be marshalled.
+type FUSEInitRes struct {
+	marshal.StubMarshallable
+
+	// InitOut contains the response from the FUSE server.
+	InitOut FUSEInitOut
+
+	// Len is the total length of bytes of the response.
+	Len uint32
+}
+
+// UnMarshalBytes deserializes src to the InitOut attribute in a FUSEInitRes.
+func (r *FUSEInitRes) UnmarshalBytes(src []byte) {
+	out := &r.InitOut
+
+	// Introduced before FUSE kernel version 7.13.
+	out.Major = uint32(usermem.ByteOrder.Uint32(src[:4]))
+	src = src[4:]
```

Comment:
Is there any particular reason to use slicing multiple times instead of a variable that traces the index or something like that?

Reply:
No particular reason it's just being consistent with the code generated by `go_marshal`. See `bazel-out/k8-fastbuild/bin/pkg/abi/linux/linux_abi_autogen_unsafe.go` in the build folder.